libs/barista-components/drawer-table/src/drawer-table.ts

```diff
@@ -0,0 +1,188 @@
+  @ContentChild(DtHeaderRowDef, { static: false }) rowHeaderDef: DtHeaderRowDef;
+
+  /** Reference to the row definition which is used to get and set the columns  */
+  @ContentChild(DtRowDef, { static: false }) rowDef: DtRowDef<T>;
+  private initialColumns: Iterable<string>;
+
+  /** @internal Used to track the currently clicked row */
+  private _currentRow: T;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+
+  get currentRow(): T {
+    return this._currentRow;
+  }
+
+  ngAfterContentInit(): void {
+    this.table.interactiveRows = true;
+  }
+
+  ngAfterViewInit(): void {
+    this.initialColumns = this.rowDef.columns;
+  }
+
+  /** Whether the drawer is open  */
+  isOpen(): boolean {
+    return !isEmpty(this.drawer) && this.drawer.opened;
+  }
+
+  /** Sets the drawer content */
+  setDrawerContent(row: T): void {
+    this.container.clear();
+    this.container.createEmbeddedView(this.drawerContentRef, {
+      $implicit: row,
+    });
+  }
+
+  /** Row click opens the drawer and sets its content or closes if the drawer is already open */
+  onRowClicked(row: T): void {
+    const open = !this.drawer.opened || this._currentRow !== row;
+    if (open) {
+      this.open();
+      this.setDrawerContent(row);
+    } else {
+      this.close();
+    }
+    this._currentRow = row;
+  }
+
+  getOpenColumns(): string[] {
+    return isEmpty(this.openColumns)
+      ? [this.initialColumns[0]]
+      : this.openColumns;
+  }
+
+  /** @internal Sets the columns of the table */
+  private _setColumns(columns: Iterable<string>): void {
+    this.rowHeaderDef.columns = columns;
+    this.rowDef.columns = columns;
+  }
+
+  /**
+   * Opens the drawer if it is not already opened.
+   */
+  open(): void {
+    this._changeDetectorRef.markForCheck();
+    this.drawer.open();
+    this._setColumns(this.getOpenColumns());
+  }
+
+  /**
+   * Closes the drawer if it is not already closed.
+   */
+  close(): void {
+    this._changeDetectorRef.markForCheck();
+    this.drawer.close();
+    this._setColumns(this.initialColumns);
+  }
+
+  /**
+   * Toggles the the drawer.
+   */
+  toggle(): void {
+    if (this.drawer.opened) {
+      this.close();
+    } else {
+      this.open();
+    }
+  }
+}
+
+/**
+ * Data row definition for the dt-table of the drawer-table component.
+ * Used to handle click event and apply style.
+ */
+@Directive({
+  selector: '[dtDrawerRowDef]',
+  exportAs: 'dtDrawerRowDef',
+  host: {
+    class: 'dt-drawer-row-def-sse',
+    '[class.dt-drawer-row-def-selected]': 'borderStyle',
+    '(click)': '_handleRowClick()',
+  },
+})
+export class DtDrawerRowDef<T> {
+  @Input('dtDrawerRowDef')
+  row: T;
+
+  constructor(private readonly drawerTable: DtDrawerTable<T>) {}
+
+  get borderStyle(): boolean {
+    if (this.drawerTable.currentRow === this.row && this.drawerTable.isOpen()) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
I believe this would be a little simpler. No need for the extra if here.
```suggestion
    return this.drawerTable.currentRow === this.row && this.drawerTable.isOpen();
```